typeagent/aitools/utils.py

```diff
@@ -299,3 +304,8 @@ def make_agent[T](cls: type[T]) -> Agent[None, T]:
         )
 
     return Agent(model, output_type=Wrapper(cls, strict=True), retries=3)
+
+
+def chunk_input(data: list[T] | str, chunk_size: int) -> Iterator[list[T] | str]:
+    for i in range(0, len(data), chunk_size):
+        yield data[i : i + chunk_size]
```

Comment:
When data is a `str`, this is going to split things in the middle of words, which is definitely causing the embedding to misunderstand the boundaries. Even if you split between words it's debatable, since you might be splitting in the middle of a key phrase or concept (e.g. "pushing up the daisies").

Reply:
Using truncation now.